core/baseline-core-v1.0-psd01.md

```diff
@@ -2792,105 +2792,156 @@ As has been done throughout this document, there are BPI layer-specific security
+
+
 #### **[D39]**	
 BPI Storage arranged in a network SHOULD utilize privacy-preserving P2P message protocols.
 
+[[D39]](#d39) Testability: Since BPI Storage arranged in a network can support pairwise key/identity relationships between storage nodes in a testable manner (see [[R282]](#r282)), and since with pairwise key/identity relationships between storage nodes any messages between storage nodes can be asymmetrically encrypted which guarantees privacy of exchanged messages between storage nodes, the requirement is testable.
+
 ## 7.3. BPI Data Orchestration
 
 To accommodate a high-volume, low latency environment with many data changes, BPI Data Orchestration has the following requirements:
 
 #### **[R289]**	
 Data Orchestration utilized in a BPI MUST NOT be a single point of failure.
 
+[[R289]](#r289) Testability: Data Orchestration solutions without a single point of failure are distributed. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
+
 #### **[R290]**	
 Data Orchestration utilized in a BPI MUST preserve source consistency. 
 
+[[R290]](#r290) Testability: Data Orchestration solutions preserving source consistency typically utilize a queued, redundant publish/subscribe messaging model. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
+
 *To avoid subscribers seeing partial and/or inconsistent data, BPI Data Orchestration has the following requirements:*
 
 #### **[R291]**	
 Data Orchestration utilized in a BPI MUST implement transaction boundaries.
 
 *This means that a single user's action can trigger atomic updates.*
 
+[[R291]](#r291) Testability: Data Orchestration solutions with transaction boundaries typically utilize topic-driven publish/subscribe messaging models. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
+
 #### **[R292]**	
-Data Orchestration utilized in a BPI MUST Commit the exact order in which operations happened on the primary database.
+Data Orchestration utilized in a BPI MUST commit the exact order in which operations happened on the primary database.
+
+[[R292]](#r292) Testability: Data Orchestration solutions which commit to and preserve operations order utilize a publish/subscribe messaging model where each message is uniquely associated to a topic and where the message identifier is deterministically created. An example of such an implementation with a complete test suite is [Apache Kafka](https://github.com/apache/kafka). Therefore, the requirement is testable.
```

Comment:
I think the logical argument here is expressed in a way that isn't easy to interpret. As I understand the point, there are data orchestration solutions that have this property (achieved in part as explained although the explanation seems insufficient). Given that it is possible to use those, and test if they are used, the requirement is testable. Am I understanding correctly?

Reply:
Yes, you understood it correctly. Why it it insufficient? @chaals ... simply stating that it is, does not make it so.

Reply:
Because deterministically creating the message identifier doesn't imply that it provides any information about ordering. I presume that the relevant information is there, based on your comments and using the example.

Reply:
> It is the combination of deterministic message identifier with a given topic Message 1,3,5,7 in My car orders the messages for my car, and 2,4,6 for My other care orders it for that topic,

Yes, along with a reconciliation system that can order those messages to create the appropriate state.

But that isn't what the text in the PR says.

Reply:
@chaals ... reconciliation is not the role of data orchestration though. The consumer pulls the messages from a topic and the order is given by the order of the message ID. If aa system then adheres to that order is a different question entirely.